### PR TITLE
ossfuzz: moving towards the ideal integration

### PR DESCRIPTION
**Pull request by cmeister2**
- Start with the basic code from the ossfuzz project.
- Rewrite fuzz corpora to be binary files full of Type-Length-Value
  data, and write a glue layer in the fuzzing function to convert
  corpora into CURL options.
- Have supporting functions to generate corpora from existing tests

This still needs integrating into the rest of the build chain properly,
but is built in CI.

### CODE REVIEW

**Comment by cmeister2** [comment]
Ok, the few tweaks I made means that this compiles within the Automake system and links to the libraries as expected (mostly this was reinstating the old code and adding some new tweaks to link to $LIB_FUZZING_ENGINE).

File: tests/fuzz/Makefile.fuzz [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Copyright 2017 Google Inc. All Rights Reserved.
```
Comment (maxdymond) [inline tests/fuzz/Makefile.fuzz]: From IRC: this file doesn't need to exist anymore now that the Makefile.am works

File: tests/fuzz/README [hunk 2]
```diff
@@ -2,13 +2,18 @@ Fuzz tests
...
+export CFLAGS="-fsanitize=address -fsanitize-address-use-after-scope -fsanitize-coverage=trace-pc-guard,trace-cmp"
+./configure --disable-shared --enable-debug --enable-maintainer-mode
 make -sj
 
 cd tests/fuzz
-make
+
+(optional) export LIB_FUZZING_ENGINE=<path to libFuzzer.a>
+
+make -f Makefile.fuzz check
```
Comment (maxdymond) [inline tests/fuzz/README]: Don't need the `-f Makefile.fuzz` qualifier here now that Automake works

**Comment by cmeister2** [comment]
As discussed on IRC, failures are due to travis issues it looks like

**Pull request closed without merging**